File: WebShopAppBackend/webshop/src/main/java/org/vs/dto/KupovinaZahtjev.java
package org.vs.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class KupovinaZahtjev {

	private String nazivKurirskeSluzbe;

	private String brojKartice;

	@NotNull
	private Integer proizvodId;

	@NotBlank
	@Pattern(regexp = "^[a-zA-Z0-9_]*$")
	@Size(min = 3, max = 100)
	private String korisnickoImeKupca;

}

File: WebShopAppFrontend/web-shop-app/src/app/model/kupovinaZahtjev.ts
export class KupovinaZahtjev {

    nazivKurirskeSluzbe: string;
    brojKartice: string;
    proizvodId: number;
    korisnickoImeKupca: string;

    constructor(
        nazivKurirskeSluzbe: string,
        brojKartice: string,
        proizvodId: number, 
        korisnickoImeKupca: string) {
        this.nazivKurirskeSluzbe = nazivKurirskeSluzbe;
        this.brojKartice = brojKartice;
        this.proizvodId = proizvodId;
        this.korisnickoImeKupca = korisnickoImeKupca;
    }

}
